### PR TITLE
Updated FieldDecl + Parser started

### PR DESCRIPTION
**Pull request by ArtInLines**
Work on parser was successfully started and the value of the FieldDecl was changed to be an Expr instead of a String. The rationale behind that is that assignments or fields can have complete Expressions and not just strings. For example:
```java
int x = 3;
int y = 3 + 4;
```
will now be parsed as
```rust
[
FieldDecl {
    ...
    val: Expr::Integer(3)
}
FieldDecl {
    ....
    val: Expr::Binary("+", Box::new(Expr::Integer(3)), Box::new(Expr::Integer(4))),
}
]
```
instead of
```rust
[
FieldDecl {
    ...
    val: String::from("3")
}
FieldDecl {
    ....
    val: String::from("3 +4") ,
}
]
```

### CODE REVIEW

**Review by Nereuxofficial — APPROVED** [review]
Looks good. Good job!

**Review by Segelente — APPROVED** [review]

File: lib/src/types.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ impl Default for Class {
 pub struct FieldDecl {
     pub field_type: Type,
     pub name: String,
-    pub val: Option<String>, // @Decide: Should probably Option<Expr> instead
+    pub val: Option<Expr>,
```
Comment (Flippchen) [inline lib/src/types.rs:43]: According to our teacher, perhaps we should consider removing the value field completely

image

Comment (ArtInLines) [reply lib/src/types.rs:43]: We need the value field if we want to support the following assignments:
```java
class A {
    int x = 9;
}
````
Without the value field, we can't represent the `= 9`. I know our teacher didn't have that in his slides, but the things in his slides were partially incomplete and partially covering things we aren't actually required to do, so I would prefer to diverge from his slides in this case.

Comment (Flippchen) [reply lib/src/types.rs:43]: Sure, I had overlooked that. Then of course we need that.

**Pull request merged**